### PR TITLE
Fixes #12318 SecurityUtils should not elminate calls to existing methods

### LINKED ISSUE #12318: SecurityUtils should not elminate calls to existing methods
**Jetty version(s)**
12.1.x

**Jetty Environment**
any

**Java version/vendor** `(use: java -version)`
any

**OS type/version**
any

**Description**
The original description below is incorrect.

The doAs problem only happens if org.eclipse.jetty.util.security.useSecurityManager is explicitly set to false on JDK17.

The doPrivileged issue and the suggested solution is still valid.

Original Description:
====================================

When "org.eclipse.jetty.util.security.useSecurityManager" is set to false, or JDK21 or later is used, then SecurityUtils treats doAs() callAs() as NOOP.

This is wrong, as both methods also perform functions unrelated to the SecurityManager (i.e. setting the subject).

The doPrivileged elimiation also seems bad , I suspect that it will lead to leaks if the SecurityManager is enabled.

In fact, I think that SecurityUtils should not try to guess the SecurityManager setting at all, but just call existing APIs if the methods exist.

The JVM will figure out the rest, and neither Jetty, nor its users have to worry about handling all the JVM/securityManager setting combinations.

**How to reproduce?**

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java [hunk 1]
```diff
@@ -93,8 +83,7 @@ public static Object getSecurityManager()
     {
         try
         {
-            if (!USE_SECURITY_MANAGER)
-                return null;
+            // TODO cache this method on class loading like the other two methods ?
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java]: let's do this

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java [hunk 2]
```diff
@@ -113,8 +102,7 @@ public static Object getSecurityManager()
      */
     public static void checkPermission(Permission permission) throws SecurityException
     {
-        if (!USE_SECURITY_MANAGER)
-            return;
+        // TODO cache this methodon class loading like the other two methods ?
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java]: lets do this

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java [hunk 3]
```diff
@@ -142,9 +130,11 @@ public static void checkPermission(Permission permission) throws SecurityExcepti
      */
     public static <T> T doPrivileged(PrivilegedAction<T> action)
     {
-        if (!USE_SECURITY_MANAGER || doPrivileged == null)
+        // Keep this method short and inlineable.
+        MethodHandle methodHandle = doPrivileged;
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java]: not sure why the local `methodHandle` variable is needed?  `doPrivileged` is final.

**Review by gregw — CHANGES_REQUESTED** [review]
I'm OK with the approach of "just call the methods if they exist and let the JVM work it out".

A couple of niggles...

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java (hunk 3 above)
Comment (stoty) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/SecurityUtils.java]: I'm not sure either, it's in 12.0 where I copied this from.
Perhaps its required for Java to automatically inline the function ? I don't know much about the automatic inlining behaviour.

I have removed it.

**Comment by stoty** [comment]
Thank you for the review, I have tried to address your comments.

Shouldn't we use CAPS for the methodHandle constants ?

**Comment by gregw** [comment]
> Shouldn't we use CAPS for the methodHandle constants ?

We have a bit of a mixed usage of CAPS for `private static final` fields, so I'd follow the style of the existing fields in this class.

**Comment by gregw** [comment]
@olamy can you look at CI failure?  Doesn't look like a flake, but then doesn't look related to this PR either.

**Review by gregw — APPROVED** [review]
This looks good, subject to CI clean build

**Comment by olamy** [comment]
@gregw build is fine
but @stoty need to sign Eclipse CLA

@stoty please have a read here https://www.eclipse.org/legal/eca/ once you have signed let us know and we will revalidate the PR

**Comment by stoty** [comment]
Thank you.
I have signed the ECA.

**Comment by olamy** [comment]
> Thank you. I have signed the ECA.

@stoty perfect. PR validated Thanks!

**Pull request merged**